src/cdsetool/download.py

```diff
@@ -158,3 +164,31 @@ def _get_credentials(options: Dict) -> Credentials:
 
 def _get_temp_dir(options: Dict) -> Union[str, None]:
     return options.get("tmpdir") or None
+
+
+def checksum_for_file(
+    file_path: str, file_name: str, proxy: Union[Dict[str, str], None] = None
+) -> bool:
+    """Evaluate the checksum of a file. If the file was downloaded correctly"""
+    odata_for_checksum = get_odata_by_name(file_name, proxy)
+    array_values = odata_for_checksum.get("value", [])
+    if len(array_values) == 0:
+        # no data available for checksum
+        return False
+    for value in array_values:
+        if value["name"] != file_name:
+            continue
+        checksums: list = value["Checksum"]
+        for checksum in checksums:
+            if checksum.get("Algorithm", "") == "MD5":
+                retrieved_value = checksum.get("Value", "")
+                calculated_value = calculate_value_checksum(file_path, "MD5")
+                return retrieved_value == calculated_value
+    assert False, f"Failed {file_name} odata does not contain any available checksum"
```

Comment:
There are some asserts in the code from before that are intended to work as a way to stop the program. The asserts don't work when running `python -O`, so we should not introduce new asserts that are essential for the program working.

I think you can just return False here to signal that the checksum check  failed.

Reply:
Ok perfect